VerbrauchZeitspanne/module.php

```diff
@@ -91,15 +120,71 @@ public function RequestAction($Ident, $Value)
          */
         public function Calculate()
         {
+            if ($this->GetStatus() != 102) {
+                return;
+            }
             $acID = IPS_GetInstanceListByModuleID('{43192F0B-135B-4CE7-A0A7-1475603F3060}')[0];
             $variableID = $this->ReadPropertyInteger('SourceVariable');
             $startDate = GetValue($this->GetIDForIdent('StartDate'));
             $endDate = GetValue($this->GetIDForIdent('EndDate'));
-
+            $levelOfDetail = $this->ReadPropertyInteger('LevelOfDetail');
+            $values = [];
             $sum = 0;
-            $values = AC_GetAggregatedValues($acID, $variableID, 1 /* Day */, $startDate, $endDate + (24 * 3600) - 1, 0);
+            if ($levelOfDetail == LOD_DATE) {
+                $values = AC_GetAggregatedValues($acID, $variableID, 1 /* Day */, $startDate, strtotime(date('d-m-Y', $endDate) . ' 23:59:59'), 0);
+            } elseif (date('d.m.Y H', $startDate) == date('d.m.Y H', $endDate)) {
```

Comment:
Gerne ein Kommentar. Was ist diese "kryptische" Prüfung?

Reply:
Außerdem müsstest du das in den Haupt else-Fall mit reinziehen, da du ja möglicherweise $startDate und $endDate anpassen musst.